Taylor/UI/Components/InsetsLabel.swift

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2016, Mirego
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// - Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+// - Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// - Neither the name of the Mirego nor the names of its contributors may
+//   be used to endorse or promote products derived from this software without
+//   specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import Foundation
+
+open class InsetsLabel: UILabel {
+    public var contentEdgeInsets: UIEdgeInsets = .zero {
+        didSet {
+            invalidateIntrinsicContentSize()
+            setNeedsDisplay()
+        }
+    }
+
+    override open var intrinsicContentSize: CGSize {
+        var size = super.intrinsicContentSize
+        size.height += contentEdgeInsets.top + contentEdgeInsets.bottom
+        return size
+    }
+
+    override open func drawText(in rect: CGRect) {
+        super.drawText(in: UIEdgeInsetsInsetRect(rect, contentEdgeInsets))
+    }
+
+    override open func sizeThatFits(_ size: CGSize) -> CGSize {
+        let baseSize = super.sizeThatFits(size)
+        return CGSize(width: size.width, height: baseSize.height + contentEdgeInsets.top + contentEdgeInsets.bottom)
```

Comment:
Pourquoi juste top et bottom? Pourquoi pas left et right?

Reply:
Parce qu'on ne veut pas que notre label devienne plus large que le width du size passé en param

Reply:
Ok, but if we see it the other way: If you want a set height and you want to get the width with the insets?

Maybe it should be: "if size.width > 0, the user wants to constraint the width, so remove the left/right when calling super, otherwise add it to the result of super; if the size.height > 0, the user wants to constraint the height, so remove the top/bottom when calling super, otherwise add to the result of super" ?

If that makes sense...

Reply:
The thing is, the default implementation of `sizeThatsFits` returns the `textRect(...)`'s size. The `textRect` is then later used to `drawText()`. Which gives us a label with a smaller width as intended, so we actually need to adjust the width of the returned size. 

As for setting the height and asking for the resulting width I tried printing `super.textRect(forBounds: CGRect(x: 0, y: 0, width: .max, height: 80), limitedToNumberOfLines: numberOfLines)` where number of line is 0
I get a result of: `(0.0, 0.0, 494.0, 22.0)`. It seems like its calculating the width with the label's font lineHeight.